pkg/clusteragent/languagedetection/patcher.go

```diff
@@ -207,33 +210,50 @@ func (lp *languagePatcher) handleDeploymentEvent(event workloadmeta.Event) {
 		if len(deployment.InjectableLanguages) > 0 {
 			// If some annotations still exist, remove them
 			annotationsPatch := lp.generateAnnotationsPatch(deployment.InjectableLanguages, langUtil.ContainersLanguages{})
-			lp.patchOwner(&owner, annotationsPatch)
-			return
+			err = lp.patchOwner(&owner, annotationsPatch)
+		} else {
+			Patches.Inc(owner.Kind, owner.Name, owner.Namespace, statusSkip)
 		}
-
-		Patches.Inc(owner.Kind, owner.Name, owner.Namespace, statusSkip)
 	} else if event.Type == workloadmeta.EventTypeSet {
 		detectedLanguages := deployment.DetectedLanguages
 		injectableLanguages := deployment.InjectableLanguages
 
 		// Calculate annotations patch
 		annotationsPatch := lp.generateAnnotationsPatch(injectableLanguages, detectedLanguages)
 		if len(annotationsPatch) > 0 {
-			lp.patchOwner(&owner, annotationsPatch)
+			err = lp.patchOwner(&owner, annotationsPatch)
 		} else {
 			Patches.Inc(owner.Kind, owner.Name, owner.Namespace, statusSkip)
 		}
 
 	}
+
+	if err != nil {
+		lp.logger.Errorf("failed to handle deployment event: %v", err)
+	}
+
 }
 
 // patches the owner with the corresponding language annotations
-func (lp *languagePatcher) patchOwner(namespacedOwnerRef *langUtil.NamespacedOwnerReference, annotationsPatch map[string]interface{}) {
+func (lp *languagePatcher) patchOwner(namespacedOwnerRef *langUtil.NamespacedOwnerReference, annotationsPatch map[string]interface{}) error {
+
+	setAnnotations := map[string]string{}
+	for k, v := range annotationsPatch {
+		if v != nil {
+			setAnnotations[k] = fmt.Sprintf("%v", v)
+		}
+	}
+
+	errs := validation.ValidateAnnotations(setAnnotations, field.NewPath("annotations"))
+
+	if len(errs) > 0 {
+		return errors.New(errs.ToAggregate().Error())
```

Comment:
Should we call Patch.Inc here with an error?

Reply:
Yes definitely, thanks for catching this.